### PR TITLE
[GLT-3711] Added dropdown filter controls

### PR DESCRIPTION
**Pull request by Arcslogger**
text-field filtering controls will be implemented in a separate PR

### CODE REVIEW

File: ts/util/table-builder.ts [hunk 1]
```diff
@@ -255,9 +373,16 @@ export class TableBuilder<ParentType, ChildType> {
...
     this.showLoading();
+    this.acceptedFilters = this.acceptedFilters.filter(
+      (filter) => filter.valid
+    );
     const response = await Rest.post(Rest.cases.query, {
       pageSize: this.pageSize,
       pageNumber: this.pageNumber,
+      filters: this.acceptedFilters.map((filter) => {
+        const mappedFilters = { key: filter.key, value: filter.value };
+        return mappedFilters;
```
Comment (djcooke) [inline ts/util/table-builder.ts:384]: Could shorten to `return { key: filter.key, value: filter.value };`

File: ts/util/table-builder.ts [hunk 2]
```diff
@@ -72,7 +124,7 @@ export class TableBuilder<ParentType, ChildType> {
 
   public build() {
     const topControlsContainer = document.createElement("div");
-    topControlsContainer.className = "flex mt-4 items-center";
+    topControlsContainer.className = "flex mt-4 items-center space-x-2";
```
Comment (djcooke) [inline ts/util/table-builder.ts:127]: Change `items-center` to `items-top` to improve display when there are enough filters to cause wrapping

File: ts/util/table-builder.ts [hunk 3]
```diff
@@ -169,10 +224,73 @@ export class TableBuilder<ParentType, ChildType> {
 
   private addFilterControls(container: HTMLElement) {
     const filterContainer = document.createElement("div");
-    filterContainer.classList.add("flex-auto");
-    container.appendChild(filterContainer);
+    filterContainer.classList.add("flex-auto", "space-x-2", "items-center");
```
Comment (djcooke) [inline ts/util/table-builder.ts:227]: Would a `space-y-2` or something improve display when wrapping happens? Currently, the +filter button can overlap with the accepted filter elements

File: ts/util/table-builder.ts [hunk 4]
```diff
@@ -33,13 +39,58 @@ export interface SortDefinition {
...
+    key: string,
+    value: string,
+    container: HTMLElement,
+    reload: () => {}
+  ) {
+    this.key = key;
+    this.value = value;
+    this.element = document.createElement("span");
+    this.element.className =
+      "font-inter font-medium text-12 text-black bg-grey-100 px-2 py-1 rounded-md cursor-default";
```
Comment (djcooke) [inline ts/util/table-builder.ts:74]: add `inline-block` to keep text together when wrapping filters

File: ts/util/table-builder.ts (hunk 3 above)
Comment (Arcslogger) [reply ts/util/table-builder.ts:227]: image
space-y-2 messes up the top margins unfortunately

Comment (djcooke) [reply ts/util/table-builder.ts:227]: Changing the accepted filter elements to inline-block actually fixes the spacing too

File: ts/util/table-builder.ts [hunk 5]
```diff
@@ -239,31 +234,31 @@ export class TableBuilder<ParentType, ChildType> {
       switch (filter.type) {
         case "dropdown":
           // create all submenu options
-          const reload = () => this.reload();
+          const onRemove = () => this.reload();
```
Comment (djcooke) [inline ts/util/table-builder.ts]: Renamed in the wrong place. Think of the different contexts:

* In `AcceptedFitler`, the function is just a callback. `AcceptedFilter` doesn't know what the function will do, so the name here should be "onRemove"
* Here, we know what the function is doing - reloading the table. It's being used when adding filters as well as removing, so "onRemove" is a misleading name here and "reload" makes sense

**Review by djcooke — APPROVED** [review]
Don't forget to create a ticket for the dropdown open/close icons

**Review by mlaszloffy — APPROVED** [review]

**Pull request merged**